README.md

```diff
@@ -1,8 +1,80 @@
 # Faithlife.Testing
 
-Home to various Faithlife testing libraries.
-
 [![Build](https://github.com/Faithlife/FaithlifeTesting/workflows/Build/badge.svg)](https://github.com/Faithlife/FaithlifeTesting/actions?query=workflow%3ABuild) [![NuGet](https://img.shields.io/nuget/v/Faithlife.Testing.svg)](https://www.nuget.org/packages/Faithlife.Testing)
 
+A collection of libraries for better testing in C#.
+
 * [Documentation](https://faithlife.github.io/FaithlifeTesting/)
 * [Release Notes](ReleaseNotes.md)
+
+## Table of Contents
+
+* [Why use Faithlife.Testing?](#why-use-faithlife.testing?)
+* [Using AssertEx.Select](#using-assertex.select)
+* [Using AssertEx.Context](#using-assertex.context)
+* [Advanced Examples](#advanced-examples)
+
+## Why use `Faithlife.Testing`?
+
+Some of the main problems that `Faithlife.Testing` is attempting to solve is:
+
+* The requirement to learn a new "syntax" when using a new test runner or assertion library.
+	```csharp
+	string actual = "test";
+	Assert.That(actual, Is.Not.Null.And.Length.AtLeast(3));
+	```
+* Many assertion libraries have common pits of failure, like using something like `Assert.IsTrue`. You're never going to get a helpful failure message out of an assertion like that.
+	```csharp
+	Assert.IsTrue(actual != null);
```

Comment:
A "fairer" comparison here would be to something like `Assert.IsTrue(actual.IsFrobbable)` -- a skeptic could say that `Assert.IsNotNull` would be the more appropriate NUnitism here.

And `AssertEx` wins that comparison *even harder* when `AssertEx.Assert(() => actual.IsFrobbable)` outputs

```
Expected:
    actual.IsFrobbable == true
    
Actual:
    actual.IsFrobbable = false
```